File: 2026/07-July/04-Saturday/solution.cs
public class Solution
{
    public int MinScore(int n, int[][] roads)
    {
        List<(int, int)>[] graph = new List<(int, int)>[n + 1];

        for (int i = 0; i <= n; i++)
            graph[i] = new List<(int, int)>();

        foreach (var road in roads)
        {
            graph[road[0]].Add((road[1], road[2]));
            graph[road[1]].Add((road[0], road[2]));
        }

        bool[] visited = new bool[n + 1];
        Queue<int> queue = new Queue<int>();

        queue.Enqueue(1);
        visited[1] = true;

        int ans = int.MaxValue;

        while (queue.Count > 0)
        {
            int node = queue.Dequeue();

            foreach (var (next, dist) in graph[node])
            {
                ans = Math.Min(ans, dist);

                if (!visited[next])
                {
                    visited[next] = true;
                    queue.Enqueue(next);
                }
            }
        }

        return ans;
    }
}

File: 2026/07-July/04-Saturday/solution.js
var minScore = function(n, roads) {
    const graph = Array.from({ length: n + 1 }, () => []);

    for (const [u, v, d] of roads) {
        graph[u].push([v, d]);
        graph[v].push([u, d]);
    }

    const visited = Array(n + 1).fill(false);
    const queue = [1];
    visited[1] = true;

    let ans = Infinity;

    while (queue.length) {
        const node = queue.shift();

        for (const [next, dist] of graph[node]) {
            ans = Math.min(ans, dist);

            if (!visited[next]) {
                visited[next] = true;
                queue.push(next);
            }
        }
    }

    return ans;
};
